### PR TITLE
electron: add breadcrumbs

### PR DESCRIPTION
**Pull request by perf2711**
This PR adds multiple new breadcrumbs to Electron main process.

Jira task: BT-776

### CODE REVIEW

File: packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        window.once('closed', () => {
+            dispose();
+            this._toDispose.splice(this._toDispose.indexOf(dispose), 1);
+        });
+    }
+
+    private listenForWindowEvents(window: BrowserWindow, breadcrumbs: BacktraceBreadcrumbs) {
+        const windowMsg = (msg: string) => `Window ${window.title} (ID: ${window.id}) ${msg}`;
+
+        const onClose = () =>
```
Comment (konraddysput) [inline packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts:45]: in this situation, we generate two logs about the same thing. Maybe it's worth marking the "onClose" event as verbose?

File: packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+        const onBlur = () =>
+            breadcrumbs.addBreadcrumb(windowMsg('lost focus'), BreadcrumbLogLevel.Info, BreadcrumbType.System);
+        const onFocus = () =>
+            breadcrumbs.addBreadcrumb(windowMsg('focused'), BreadcrumbLogLevel.Info, BreadcrumbType.System);
+        const onShow = () =>
+            breadcrumbs.addBreadcrumb(windowMsg('shown'), BreadcrumbLogLevel.Info, BreadcrumbType.System);
+        const onHide = () =>
+            breadcrumbs.addBreadcrumb(windowMsg('hidden'), BreadcrumbLogLevel.Info, BreadcrumbType.System);
+        const onReadyToShow = () =>
```
Comment (konraddysput) [inline packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts:77]: shouldn't this event be verbose?

**Review by konraddysput — APPROVED** [review]

File: packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts (hunk 1 above)
Comment (perf2711) [reply packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts:45]: Changed.

File: packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts (hunk 2 above)
Comment (perf2711) [reply packages/electron/src/main/breadcrumbs/WindowEventSubscriber.ts:77]: Changed.

**Review by konraddysput — APPROVED** [review]

**Pull request merged**